pip/src/interpreter.rs

```diff
@@ -27,11 +31,33 @@ use qsc::{
     target::Profile,
     LanguageFeatures, PackageType, SourceMap,
 };
+use qsc_qasm3::{OutputSemantics, ProgramType};
 use resource_estimator::{self as re, estimate_expr};
 use std::{cell::RefCell, fmt::Write, path::PathBuf, rc::Rc};
 
+/// If the classes are not Send, the Python interpreter
+/// will not be able to use them in a separate thread.
+///
+/// This function is used to verify that the classes are Send.
+/// The code will fail to compile if the classes are not Send.
+fn verify_classes_are_sendable() {
+    fn is_send<T: Send>() {}
+    is_send::<TargetProfile>();
+    is_send::<Result>();
+    is_send::<Pauli>();
+    is_send::<Output>();
+    is_send::<StateDumpData>();
+    is_send::<Circuit>();
+
+    // QSharpError, and QasmError are not Send, but we don't raise
```

Comment:
But we do raise `QSharpError` in this module. What am I missing?

Reply:
We return `QasmError::new_err` or `QSharpError::new_err` which actually returns a `PyErr` that is `Send` and the args passed into the `new_err` call must also impl `Send` because of this.